_test/check_exercises.nim

```diff
@@ -90,12 +90,10 @@ proc getImplementedSlugs: Slugs =
   ## Let us consider an "implemented exercise" as one with a correctly named
   ## test file, rather than one with an entry in `config.json`. This can be more
   ## convenient when implementing new exercises.
-  for dir in walkDirs(exercisesDir / "concept" / "*"):
-    for file in walkFiles(dir / "test_*.nim"):
-      result.incl(dir.splitPath().tail, ekConcept) # e.g. "hello-world"
-  for dir in walkDirs(exercisesDir / "practice" / "*"):
-    for file in walkFiles(dir / "test_*.nim"):
-      result.incl(dir.splitPath().tail, ekPractice) # e.g. "hello-world"
+  for ek in [ekConcept, ekPractice]:
```

Comment:
Good suggestion - the iterator does produce smaller C output.

But I believe C compilers will compile both to the same thing when compiling with optimization switches. So I have a slight preference for the array, because it's clearer that we're only iterating over two things. So I'll merge it like that.

---

If we consider the Nim code

```Nim
type
  Foo = enum
    fAaa, fBbb, fCcc

proc main =
  # for f in fBbb..fCcc:
  for f in [fBbb, fCcc]:
    echo "hi"

when isMainModule:
  main()
```

both produce (compiling with `-d:release`)

```
PreMainInner:
 endbr64 
 ret    
 data16 nop WORD PTR cs:[rax+rax*1+0x0]
NimMainInner:
 endbr64 
 sub    rsp,0x8
 mov    esi,0x1
 lea    rdi,[rip+0x0]        # 24 <NimMainInner+0x14>
 call   29 <NimMainInner+0x19>
 mov    esi,0x1
 lea    rdi,[rip+0x0]        # 35 <NimMainInner+0x25>
 add    rsp,0x8
 jmp    3e <NimMainInner+0x2e>
 xchg   ax,ax
main_example_8:
 endbr64 
 sub    rsp,0x8
 mov    esi,0x1
 lea    rdi,[rip+0x0]        # 54 <main_example_8+0x14>
 call   59 <main_example_8+0x19>
 mov    esi,0x1
 lea    rdi,[rip+0x0]        # 65 <main_example_8+0x25>
 add    rsp,0x8
 jmp    6e <main_example_8+0x2e>
 xchg   ax,ax
PreMain:
 endbr64 
 sub    rsp,0x18
 mov    rax,QWORD PTR fs:0x28
 mov    QWORD PTR [rsp+0x8],rax
 xor    eax,eax
 lea    rax,[rip+0x0]        # 8f <PreMain+0x1f>
 mov    QWORD PTR [rsp],rax
 call   98 <PreMain+0x28>
 mov    rdi,rsp
 call   a0 <PreMain+0x30>
 call   a5 <PreMain+0x35>
 mov    rax,QWORD PTR [rsp]
 call   rax
 mov    rax,QWORD PTR [rsp+0x8]
 xor    rax,QWORD PTR fs:0x28
 jne    c0 <PreMain+0x50>
 add    rsp,0x18
 ret    
 call   c5 <PreMain+0x55>
 data16 nop WORD PTR cs:[rax+rax*1+0x0]
NimMain:
 endbr64 
 sub    rsp,0x28
 mov    rax,QWORD PTR fs:0x28
 mov    QWORD PTR [rsp+0x18],rax
 xor    eax,eax
 lea    rax,[rip+0x0]        # ef <NimMain+0x1f>
 mov    QWORD PTR [rsp+0x10],rax
 call   f9 <NimMain+0x29>
 lea    rdi,[rsp+0x10]
 call   103 <NimMain+0x33>
 call   108 <NimMain+0x38>
 mov    rax,QWORD PTR [rsp+0x10]
 call   rax
 lea    rax,[rip+0x0]        # 116 <NimMain+0x46>
 lea    rdi,[rsp+0x8]
 mov    QWORD PTR [rsp+0x8],rax
 call   125 <NimMain+0x55>
 mov    rax,QWORD PTR [rsp+0x8]
 call   rax
 mov    rax,QWORD PTR [rsp+0x18]
 xor    rax,QWORD PTR fs:0x28
 jne    141 <NimMain+0x71>
 add    rsp,0x28
 ret    
 call   146 <NimMain+0x76>
 nop    WORD PTR cs:[rax+rax*1+0x0]
NimMainModule:
 endbr64 
 sub    rsp,0x8
 mov    esi,0x1
 lea    rdi,[rip+0x0]        # 164 <NimMainModule+0x14>
 call   169 <NimMainModule+0x19>
 mov    esi,0x1
 lea    rdi,[rip+0x0]        # 175 <NimMainModule+0x25>
 add    rsp,0x8
 jmp    17e <NimMainModule+0x2e>
main:
 endbr64 
 sub    rsp,0x8
 mov    QWORD PTR [rip+0x0],rsi        # f <main+0xf>
 mov    DWORD PTR [rip+0x0],edi        # 15 <main+0x15>
 mov    QWORD PTR [rip+0x0],rdx        # 1c <main+0x1c>
 call   21 <main+0x21>
 mov    eax,DWORD PTR [rip+0x0]        # 27 <main+0x27>
 add    rsp,0x8
 ret
```